packages/components/src/Modal/GenericModal/GenericModal.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { HTMLAttributes, useId, useState } from "react"
+import { createPortal } from "react-dom"
+import { Transition } from "@headlessui/react"
+import FocusLock from "react-focus-lock"
+import { warn } from "~utils/console"
+import { ModalContext, ModalContextType } from "./subcomponents/ModalContext"
+import styles from "./GenericModal.module.scss"
+
+export type GenericModalProps = {
+  id?: string
+  isOpen: boolean
+  children: React.ReactNode
+  focusLockDisabled?: boolean
+  onEscapeKeyup?: (event: KeyboardEvent) => void
+  onOutsideModalClick?: (event: React.MouseEvent) => void
+  onAfterLeave?: () => void
+} & Omit<ModalContextType, "labelledByID" | "describedByID"> &
```

Comment:
Looks like you Omitted everything you added :P

```suggestion
} &
```